### PR TITLE
Address missing rendering pieplines for translucent and simplify Model rendering code

### PR DESCRIPTION
**Pull request by samcake**
- If FBX material has a 0 opacity, force it to 1 since 0 means invisible, we 'll assume that's not what's expected, we could argue that 0 Opacity means that would should still see the specular/reflection. Let's consider that when transparent works correctly...
- Replace all the explicit lists of bucketing in Model by a HashMap using the REnderKey also used for the RenderPIpeline.
- Add Nsight support to Render-Utils
- Add missing RenderPIpeline for Translucent and normals/specular/skinned enven thrugh we don;t have a good rendering path for it yet

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f06556ba1245bebcbb11414f63255f5c896b1acc available at: https://gist.github.com/c743262753d557482669<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4657.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4657.dmg

**Comment by OmegaHeron** [comment]
I like this one a  lot - Blender has a reversed notion in its internal render where you set alpha to 0 then have image texture influence that.  If you have it set to 1 it shows as solid regardless of influence.  If you forget to change it to 1.0 with texture influence at 1.0, HF formerly, rendered as 100% transparent regardless of texture alpha.  Just tested this and it works will all current content in place with alpha textures and an import with it done the "wrong" way works as well.

**Comment by samcake** [comment]
Thanks !
I didn't know about that behavior from Blender.
Glad it works

S

File: libraries/render-utils/src/Model.h [hunk 1]
```diff
@@ -527,6 +486,22 @@ class Model : public QObject, public PhysicsEntity {
...
+        QVector<int> _meshes;
+        QMap<QString, int> _unsortedMeshes;
+    };
+    typedef std::unordered_map<int, RenderBucket> BaseRenderBucketMap;
+    class RenderBucketMap : public BaseRenderBucketMap {
+    public:
+        typedef RenderKey Key;
+
+        
+        void addRenderBucket(Key key, RenderBucket& bucket);
```
Comment (Atlante45) [inline libraries/render-utils/src/Model.h]: It looks like this method is not implemented.

**Comment by cozza13** [comment]
Just tested this on Mac and it is fantastic. Can we get this merged

**Comment by samcake** [comment]
THis should be ready for merge

**Comment by hifi-gustavo** [comment]
Build console output for commit: edb388ed834b2c6ab5b2c22f6fd6f5c5f3429959 available at: https://gist.github.com/c743262753d557482669<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4657.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4657.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 525d1040edfd408784d2765abbb45cc22faf3104 available at: https://gist.github.com/c743262753d557482669<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4657.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4657.dmg

**Pull request merged**